test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1

```diff
@@ -131,12 +131,252 @@ Describe "Validate Registry Provider" -Tags @('CI', 'RequireAdminOnWindows') {
             $properties = @(Get-ItemProperty -Path $testKey)
             $properties.Count | Should Be 0
         }
+    }
+}
+
+Describe "Extended Registry Provider Tests" -Tags @("Feature", "RequireAdminOnWindows") {
+    BeforeAll {
+        $registryBase = "HKLM:\software\Microsoft\PowerShell\3\"
+        $testKey = "TestKeyThatWillNotConflict"
+        $testSubKey = "SubKey"
+        $testSubKey2 = "SubKey2"
+        $restoreLocation = Get-Location
+        $testPropertyName = "TestEntry"
+        $testPropertyValue = 1
+
+        #skip all tests on non-windows platform
+        $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+        if ($IsWindows -eq $false) {
+            $PSDefaultParameterValues["it:skip"] = $true
+        }
+    }
+
+    AfterAll {
+        #restore the previous environment
+        Set-Location -Path $restoreLocation
+        $global:PSDefaultParameterValues = $originalDefaultParameterValues
+    }
+
+    BeforeEach {
+        Set-Location $registryBase
+        New-Item -Path $testKey > $null
+        Set-Location $testKey
+        New-Item -Path $testSubKey > $null
+        New-Item -Path $testSubKey2 > $null
+        New-ItemProperty -Path $testSubKey -Name $testPropertyName -Value $testPropertyValue -PropertyType DWORD > $null
+    }
+
+    AfterEach {
+        Set-Location $registryBase
+        Remove-Item -Path $testKey -Recurse -Force -ErrorAction SilentlyContinue
+    }
+
+    Context "Valdiate New-ItemProperty Parameters" {
+        BeforeEach {
+            Remove-ItemProperty -Path $testSubKey -Name $testPropertyName -Force -ErrorAction SilentlyContinue
+        }
+
+        It "Verify Filter" {
+            try {
+                $result = New-ItemProperty -Path ".\*" -Filter "*Test*" -Name $testPropertyName -Value $testPropertyValue -PropertyType DWORD -ErrorAction Stop
+                throw "Expected exception not thrown"
+            }
+            catch { $_.FullyQualifiedErrorId | Should Be "NotSupported,Microsoft.PowerShell.Commands.NewItemPropertyCommand" }
+        }
+
+        It "Verify Include" {
+            $result = New-ItemProperty -Path ".\*" -Include "*2" -Name $testPropertyName -Value $testPropertyValue -PropertyType DWORD
+            $result."$testPropertyName" | Should Be $testPropertyValue
```

Comment:
TL;DR Fixed.
You are right, I applied a path validation to all of the tests that I think it applies. This also found errors in my code so there are refactors in those areas as well.